flipperdevices/flipperzero-good-faps
# Fix FlipTDI crashing on Linux when connecting to a Linux system

# What's new

- fixes crash in `ftdi_usb_callback_tx_immediate` due to assert

I'm not entirely sure how accurate my debugging was, the debugger kept pointing at `furi_thread_get_id(ftdi_usb->thread)` in the traceback and looking through the short `furi_thread_get_id` code shows that its checking the given pointer using `furi_check` and apparently thats where an assert was triggering.

I haven't managed to inspect the contents of ftdi_usb because it was optimized out and a log statement didn't print anything.

# Verification 

- Start FlipTDI
- Plug into Linux system
- verify the device is seen as an FT232H using `dmesg`
- exit app

# Checklist (For Reviewer)
